crates/daphne-worker/src/aggregator/roles/leader.rs

```diff
@@ -0,0 +1,281 @@
+// Copyright (c) 2024 Cloudflare, Inc. All rights reserved.
+// SPDX-License-Identifier: BSD-3-Clause
+
+use std::borrow::Cow;
+
+use crate::{aggregator::App, elapsed};
+use axum::{async_trait, http::Method};
+use daphne::{
+    constants::{DapMediaType, DapRole},
+    error::DapAbort,
+    fatal_error,
+    messages::{BatchId, BatchSelector, Collection, CollectionJobId, Report, TaskId},
+    roles::{leader::WorkItem, DapAggregator as _, DapLeader},
+    DapAggregationParam, DapCollectionJob, DapError, DapRequestMeta, DapResponse, DapVersion,
+};
+use daphne_service_utils::http_headers;
+use http::{header, HeaderMap, HeaderName, HeaderValue, StatusCode};
+use prio::codec::ParameterizedEncode;
+use tracing::{error, info};
+use url::Url;
+
+#[async_trait]
+impl DapLeader for App {
+    async fn put_report(&self, report: &Report, task_id: &TaskId) -> Result<(), DapError> {
+        let task_config = self
+            .get_task_config_for(task_id)
+            .await?
+            .ok_or(DapAbort::UnrecognizedTask { task_id: *task_id })?;
+
+        worker::console_log!("{:?}", &*self.test_leader_state.lock().unwrap());
+        self.test_leader_state
+            .lock()
+            .unwrap()
```

Comment:
It seems odd to grab the lock twice. In theory, some other process could change `self.test_leader_state` between the call to `console_log!()` and when we actually call `put_report()`.

I'd suggest just grabbing the lock once.

Here and below.

Reply:
This was actually leftover logs for debugging